js/components/main.js

```diff
@@ -429,6 +459,28 @@ class Main extends ImmutableComponent {
     contextMenus.onTabsToolbarContextMenu(activeFrame, undefined, undefined, e)
   }
 
+  get allSiteSettings () {
+    const activeFrame = FrameStateUtil.getActiveFrame(this.props.windowState)
+    if (activeFrame && activeFrame.get('isPrivate')) {
+      return this.props.appState.get('temporarySiteSettings')
+    }
+    return this.props.appState.get('siteSettings')
+  }
+
+  get activeSiteSettings () {
+    const activeFrame = FrameStateUtil.getActiveFrame(this.props.windowState)
+    if (!activeFrame) {
+      return undefined
+    }
+    return siteSettings.getSiteSettingsForURL(this.allSiteSettings, activeFrame.get('location'))
+  }
+
+  get braveShieldsDisabled () {
+    const activeFrame = FrameStateUtil.getActiveFrame(this.props.windowState)
+    const parsedUrl = urlParse(activeFrame && activeFrame.get('location') || '')
+    return !(parsedUrl.protocol || '').startsWith('http')
```

Comment:
highly unlikely, but this would return true for custom schemes that happen to start with `http`